### PR TITLE
Update API

### PR DESCRIPTION
**Pull request by oded-dd**

### CODE REVIEW

File: v1/Probe.dhall [hunk 1]
```diff
@@ -0,0 +1,12 @@
+let Kubernetes = (../imports.dhall).Kubernetes
+
+let ProbeSpec = ./ProbeSpec.dhall
+
+in  { Type =
+        { apiVersion : Text
+        , kind : Text
+        , metadata : Kubernetes.ListMeta.Type
```
Comment (ari-becker) [inline v1/Probe.dhall:8]: a `Probe` has a `metadata` of type `ListMeta`?

File: v1/StorageSpec.dhall [hunk 2]
```diff
@@ -4,10 +4,39 @@ let Kubernetes = imports.Kubernetes
...
+      Common ⩓ { volumeClaimTemplate : EmbeddedPersistentVolumeClaim.Type }
+
+let common = { disableMountSubPath = None Bool }
+
 let StorageSpec =
-      < EmptyDir : { emptyDir : Kubernetes.EmptyDirVolumeSource.Type }
-      | VolumeClaimTemplate :
-          { volumeClaimTemplate : EmbeddedPersistentVolumeClaim.Type }
-      >
+      { Union =
```
Comment (ari-becker) [inline v1/StorageSpec.dhall:17]: If you're going to make this change then I prefer that it be called `Type` not `Union`

File: v1/ProbeSpec.dhall [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        , targets : Optional ProbeTargets.Type
+        , interval : Optional Text
+        , scrapeTimeout : Optional Text
+        }
+    , default =
+        { jobName = None Text
+        , module = None Text
+        , interval = None Text
+        , scrapeTimeout = None Text
+        }
```
Comment (ari-becker) [inline v1/ProbeSpec.dhall:18]: missing `prober`, `interval`

File: v1/ProbeTargetIngress.dhall [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+let NamespaceSelector = ./NamespaceSelector.dhall
+
+let RelabelConfig = ./RelabelConfig.dhall
+
+in  { Type =
+        { selector : Optional Kubernetes.LabelSelector.Type
+        , namespaceSelector : Optional NamespaceSelector
+        , relabelingConfigs : Optional (List RelabelConfig.Type)
+        }
+    , default = {=}
```
Comment (ari-becker) [inline v1/ProbeTargetIngress.dhall:14]: Missing `selector`, `namespaceSelector`, `relabelingConfigs`

File: v1/ProbeTargetStaticConfig.dhall [hunk 5]
```diff
@@ -0,0 +1,7 @@
+let imports = ../imports.dhall
+
+let Map = imports.Prelude.Map.Type
+
+in  { Type =
+        { static : Optional (List Text), labels : Optional (Map Text Text) }
+    }
```
Comment (ari-becker) [inline v1/ProbeTargetStaticConfig.dhall:7]: missing `default` with `static`, `labels`

File: v1/ProbeTargets.dhall [hunk 6]
```diff
@@ -0,0 +1,9 @@
+let ProbeTargetStaticConfig = ./ProbeTargetStaticConfig.dhall
+
+let ProbeTargetIngress = ./ProbeTargetIngress.dhall
+
+in  { Type =
+        { staticConfig : Optional ProbeTargetStaticConfig.Type
+        , ingress : Optional ProbeTargetIngress.Type
+        }
+    }
```
Comment (ari-becker) [inline v1/ProbeTargets.dhall:9]: missing `default` with `staticConfig`, `ingress`

File: v1/ProberSpec.dhall [hunk 7]
```diff
@@ -0,0 +1,3 @@
+{ Type = { url : Text, scheme : Optional Text, path : Optional Text }
+, default = { scheme = Some "http", path = Some "/probe" }
```
Comment (ari-becker) [inline v1/ProberSpec.dhall:2]: Should be `None Text` for both `scheme`, `path` (if you want to document that `None Text` is equivalent to `http` or `/probe`, then put it in a comment)

File: v1/StorageSpec.dhall [hunk 8]
```diff
@@ -4,10 +4,39 @@ let Kubernetes = imports.Kubernetes
...
 let StorageSpec =
-      < EmptyDir : { emptyDir : Kubernetes.EmptyDirVolumeSource.Type }
-      | VolumeClaimTemplate :
-          { volumeClaimTemplate : EmbeddedPersistentVolumeClaim.Type }
-      >
+      { Type =
+          < EmptyDir : EmptyDir | VolumeClaimTemplate : VolumeClaimTemplate >
+      , EmptyDir = { Type = EmptyDir, default = common }
+      , VolumeClaimTemplate = { Type = VolumeClaimTemplate, default = common }
+      }
```
Comment (ari-becker) [inline v1/StorageSpec.dhall:21]:
```dhall
let Options = { EmptyDir , VolumeClaimTemplate }
let StorageSpec = 
          < EmptyDir : Options.EmptyDir | VolumeClaimTemplate : Options.VolumeClaimTemplate >
in    { Type = StorageSpec
      , EmptyDir = \(empty-dir : Options.EmptyDir.Type) -> StorageSpec.EmptyDir empty-dir
      , VolumeClaimTemplate = \(volume-claim-template : Options.VolumeClaimTemplate) -> StorageSpec.VolumeClaimTemplate volume-claim-template
      , Options
      }
```

File: v1/ThanosRulerStatus.dhall [hunk 9]
```diff
@@ -0,0 +1,10 @@
+{ Type =
+  { paused : Bool
+    , replicas : Natural
+      , updatedReplicas : Natural
+      , availableReplicas : Natural
+      , unavailableReplicas : Natural
+  }
+  , default = {=}
+}
+
```
Comment (ari-becker) [inline v1/ThanosRulerStatus.dhall:10]: Not formatted

File: .github/CODEOWNERS [hunk 10]
```diff
@@ -1 +1 @@
-* @ari-becker @oded-dd
+* @ari-becker @oded-dd @ShauliSolomovich
```
Comment (ari-becker) [inline .github/CODEOWNERS:1]: add Shauli to the README as a Maintainer

File: v1/StorageSpec.dhall (hunk 8 above)
Comment (oded-dd) [reply v1/StorageSpec.dhall:21]: I am not in favor of that, declining it.

**Pull request merged**